cmd/trace-agent/config/config.go

```diff
@@ -75,17 +75,12 @@ func prepareConfig(path string) (*config.AgentConfig, error) {
 	cfg.DDAgentBin = defaultDDAgentBin
 	cfg.AgentVersion = version.AgentVersion
 	cfg.GitCommit = version.Commit
-	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
-	orch := fargate.GetOrchestrator(ctx)
-	cancel()
-	if err := ctx.Err(); err != nil && err != context.Canceled {
-		log.Errorf("Failed to get Fargate orchestrator. This may cause issues if you are in a Fargate instance: %v", err)
-	}
-	cfg.FargateOrchestrator = config.FargateOrchestratorName(orch)
 	coreconfig.Datadog.SetConfigFile(path)
 	if _, err := coreconfig.Load(); err != nil {
 		return cfg, err
 	}
+	orch := fargate.GetOrchestrator() // Needs to be after loading config, because it relies on feature auto-detection
```

Comment:
This comment is good to make sure we don't accidentally break this with re-ordering but I'm wondering if we can also add or modify a unit test here to better enforce this? (I only ask this since we've had a breaking change in this exact area in the past  slip through because it needed to be _after_ config file loading but there were no tests that checked it explicitly)

Reply:
Hi @ajgajg1134 

here is a unit test to validate Fargate orchestrator detection in trace-agent config: https://github.com/DataDog/datadog-agent/pull/14746/commits/224656f81f36c519805f05b68c861e2baf285881